File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mpp;
        for(int i=0; i<nums.size(); i++){
            int toFind = target - nums[i];

            if (mpp.find(toFind) != mpp.end()){
                return {mpp[toFind], i};
            }

            mpp[nums[i]] = i;
        }

        return {-1,-1};
    }
};

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    seen := map[int]int{}
    for i, num := range nums {
        toFind := target - num

        if j, ok := seen[toFind]; ok {
            return []int{j, i}
        }

        seen[num] = i
    }

    return []int{}
}